easylinks/src/main/java/com/google/step/servlets/EasyLinkServlet.java

```diff
@@ -7,33 +7,30 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet responses to redirect the easy link to another website*/
-@WebServlet("/easy-link")
+@WebServlet("/*")
 public class EasyLinkServlet extends HttpServlet {
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String easyLink = ServletHelper.getParameterWithDefault(request, "easy-link", "");
-
-    if (easyLink.isEmpty()) {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String requestUrl = request.getRequestURI();
+    if (requestUrl.isEmpty()) {
+      // Invalid input url, redirect to the home page
       response.sendRedirect(HOME_PAGE);
-    } else if (easyLink.startsWith(NAVIGATION_LINK)) {
-      // TODO: Search the datastore to find the link
-    } else if (easyLink.startsWith(PEOPLE_SEARCH_LINK)) {
-      // TODO: Navigate
-    } else if (easyLink.startsWith(PROVIDED_LINK)) {
-      // TODO: Search the people
-    } else {
-      // Process the cutomized link and redirect
-      String url = ServletHelper.fetchUrlWithDefault(
-                    easyLink, ServletHelper.USERSERVICE.getCurrentUser().getEmail(), HOME_PAGE);
-      response.sendRedirect(url);
+      return;
     }
     
+    String responseUrl = ServletHelper.fetchUrlWithDefault(requestUrl.substring(1),
+                            ServletHelper.USERSERVICE.getCurrentUser().getEmail(), HOME_PAGE);
+    response.sendRedirect(responseUrl);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Redirect to the corresponding servlet
+    String shortcut = ServletHelper.getParameterWithDefault(request, "easy-link", HOME_PAGE);
+    response.sendRedirect(shortcut);
```

Comment:
I'm not sure if you tested, do we need to add "/" before shortcut?

Reply:
For the post which is the input get from UI, no need to add "/".
For the url bar input, like xxx.com/rmp, we need to put /rmp.

Reply:
We can probably add the "/" in the input box and make it irremovable.

Reply:
Do you mean the input box in the UI? That doesn't matter, because the input box in the UI will post to the servlet and the servlet will redirect to another one. When redirecting, "/" will be added automatically.

Reply:
Oh I see.